scripts/us_epa/parent_company/process_parent_company.py

```diff
@@ -0,0 +1,298 @@
+flags.DEFINE_string("input_download_path", "tmp_data", "Input directory")
+flags.DEFINE_string("existing_facilities_file", "existing_facilities",
+                    "Filename for existing facilities ids")
+flags.DEFINE_string("output_base_path", "output",
+                    "Output directory for processed data.")
+flags.DEFINE_string("parent_co_output_path", "table",
+                    "Output directory for company info.")
+flags.DEFINE_string("ownership_output_path", "ownership",
+                    "Output directory for ownership.")
+
+# V_PARENT_COMPANY_INFO table
+_TABLE_PREFIX = "D_GHG_B"
+_TABLE = "V_PARENT_COMPANY_INFO"
+
+_OUT_FILE_PREFIX = "EpaParentCompany"
+_COUNTY_CANDIDATES_CACHE = {}
+
+# Cleaned CSV Columns
+# - "containedInPlace" is a repeated list of refs to County and Census ZCTA
+_DCID = "dcid"
+_EPA_FACILITY_GHG_ID = "epaGhgrpFacilityId"
+_NAME = "parentCompanyName"
+_OTHERNAME = "otherName"
+_YEAR = "year"
+_PERCENT_OWNERSHIP = "facilityPercentOwnership"
+_ADDRESS = "address"
+_CIP = "locatedIn"
+
+_TABLE_CLEAN_CSV_HDR = (_DCID, _NAME, _ADDRESS, _CIP)
+_OWNERSHIP_CLEAN_CSV_HDR = (_DCID, _EPA_FACILITY_GHG_ID, _YEAR,
+                            _PERCENT_OWNERSHIP)
+
+_COUNTERS = {
+    "missing_zip": set(),
+    "percent_ownership_not_found": set(),
+    "facility_does_not_exist": set(),
+    "company_name_not_found": set(),
+    "year_does_not_exist": set(),
+}
+
+
+def _gen_mcf():
+    lines = [
+        "Node: dcid:EpaOrganizationOwnership",
+        "description: The ownership of an EPA Facility by an Organization in a given year.",
+        "typeOf: dcs:StatisticalVariable", "populationType: dcs:Organization",
+        "measuredProperty: owns", "statType: measurementResult"
+    ]
+    return "\n".join(lines)
+
+
+def _gen_ownership_tmcf():
+    lines = [
+        "Node: E:EpaParentCompanyOwnership->E0",
+        "typeOf: dcs:StatVarObservation",
+        "variableMeasured: dcid:EpaOrganizationOwnership",
+        f"observationDate: C:EpaParentCompanyOwnership->{_YEAR}",
+        f"observationAbout: C:EpaParentCompanyOwnership->{_DCID}",
+        f"value: C:EpaParentCompanyOwnership->{_EPA_FACILITY_GHG_ID}",
+        f"ownershipPercentage: C:EpaParentCompanyOwnership->{_PERCENT_OWNERSHIP}",
+    ]
+    return "\n".join(lines)
+
+
+def _gen_company_tmcf():
+    lines = [
+        "Node: E:EpaParentCompanyTable->E1", "typeOf: dcs:Organization",
+        f"{_DCID}: C:EpaParentCompanyTable->{_DCID}",
+        f"{_NAME}: C:EpaParentCompanyTable->{_NAME}",
+        f"{_ADDRESS}: C:EpaParentCompanyTable->{_ADDRESS}",
+        f"{_CIP}: C:EpaParentCompanyTable->{_CIP}"
+    ]
+    return "\n".join(lines)
+
+
+def _v(table, row, col):
+    return row.get(_TABLE_PREFIX + "." + table + "." + col, "")
+
+
+def _cv(table, row, col):
+    return _v(table, row, col).strip().title()
+
+
+def _get_name(table, row):
+    name = _cv(table, row, "PARENT_COMPANY_NAME")
+    return name.replace(" Llc", " LLC")
+
+
+def _name_to_id(s):
```

Comment:
i'm curious if you did some quick slices (by state, etc) to see if this id scheme produced unique, correct, id's and didn't end up merging companies.

Reply:
Yup! Please see the conversation separate in chat.